backend/tests/settings/pure_variants/test_pure_variants_license_routes.py

```diff
@@ -0,0 +1,118 @@
+# SPDX-FileCopyrightText: Copyright DB InfraGO AG and contributors
+# SPDX-License-Identifier: Apache-2.0
+
+import io
+
+import pytest
+from fastapi import testclient
+from kubernetes import client as kubernetes_client
+
+
+@pytest.mark.usefixtures("admin", "pure_variants_license")
+def test_get_license(client: testclient.TestClient):
+    response = client.get(
+        "/api/v1/settings/integrations/pure-variants",
+    )
+
+    assert response.is_success
+    assert response.json()["license_server_url"] == "http://127.0.0.1:27000"
+    assert response.json()["license_key_filename"] == "xy.license"
+
+
+@pytest.mark.usefixtures("admin")
+def test_set_license_if_unset(client: testclient.TestClient):
+    """Set the p::v license the first time"""
+
+    response = client.patch(
+        "/api/v1/settings/integrations/pure-variants",
+        json={
+            "license_server_url": "http://localhost:27000",
+        },
+    )
+
+    assert response.is_success
+    assert response.json()["license_server_url"] == "http://localhost:27000"
+
+
+@pytest.mark.usefixtures("admin", "pure_variants_license")
+def test_set_license_if_set_already(client: testclient.TestClient):
+    """Update an existing license"""
+
+    response = client.patch(
+        "/api/v1/settings/integrations/pure-variants",
+        json={
+            "license_server_url": "http://localhost:27000",
+        },
+    )
+
+    assert response.is_success
+    assert response.json()["license_server_url"] == "http://localhost:27000"
```

Comment:
```suggestion
@pytest.mark.usefixtures("admin", "pure_variants_license")
def test_set_license_if_set_already(client: testclient.TestClient):
    """Update an existing license"""

    prev_license_server_url = pure_variants_license.license_server_url

    response = client.patch(
        "/api/v1/settings/integrations/pure-variants",
        json={
            "license_server_url": "http://localhost:27000",
        },
    )

    pure_variants_config = pure_variants_crud.get_pure_variants_configuration(
        db
    )

    assert pure_variants_config
    assert pure_variants_config.license_server_url == "http://localhost:27000"

    assert prev_license_server_url != pure_variants_config.license_server_url

    assert response.is_success
    assert response.json()["license_server_url"] == "http://localhost:27000"
```
For me this test here does not completely check the update process as it is.

1. I would always include some check for the precondition. In this case, I would check that the previous license server url was other than `http://localhost:27000`
2. I would include a check that the value was actually changed in the database, i.e., not only returned by the route
3. Personally, I would always either use `localhost` or `127.0.0.1` and just adjust the port, i.e., from `27000` to `28000` as such a change is easier to see for me. However, if for you the current naming works better than we can also keep it